src/fold.c

```diff
@@ -2911,7 +2911,8 @@ foldSplit(
     // any between top and bot, they have been removed by the caller.
     gap1 = &fp->fd_nested;
     gap2 = &fp[1].fd_nested;
-    if (foldFind(gap1, bot + 1 - fp->fd_top, &fp2))
+    (void)foldFind(gap1, bot + 1 - fp->fd_top, &fp2);
+    if (fp2 != NULL)
```

Comment:
You should add the below test to test_fold.vim for this.

func Test_fold_split()
  new
  let lines =<< trim END
    line 1
      line 2
      line 3
        line 4
        line 5
  END
  call setline(1, lines)
  setlocal sw=2
  setlocal foldmethod=indent foldenable
  call assert_equal([0, 1, 1, 2, 2], range(1, 5)->map('foldlevel(v:val)'))
  call append(2, 'line 2.5')
  call assert_equal([0, 1, 0, 1, 2, 2], range(1, 6)->map('foldlevel(v:val)'))
  bw!
endfunc

Reply:
I would like to work on adding that test.

I haven't used these test files before, though. How do I run one function or all the functions within src/testdir/test_fold.vim?

Thanks!

Reply:
Nevermind. I found the README.

Reply:
The instructions for running the tests are in the testdir/README.txt file.

From the testdir directory, you can run the `make test_fold` command to execute
the tests in the test_fold.vim file.

To execute only the Test_fold_split() function, you can run the 
`../vim -u NONE -S runtest.vim test_fold.vim Test_fold_split` command
from the testdir directory.

If the test fails, then the test.log file will contain the errors. Otherwise, this file
will not be present. You can check the 'messages' file for the messages from
the test.

Reply:
Thank you for the instructions (and the test itself)!
I've added the test and made sure it fails before my fix and succeeds after my fix.